File: java/src/peachtree/aln/Taxon.java
package peachtree.aln;

public class Taxon {

	int id;
	String name;
	boolean isSelected; // Is the taxon selected
	boolean isHighlighted;
	
	public Taxon(int id, String name) {
		this.id = id;
		this.name = name;
		this.isHighlighted = false;
		this.isSelected = false;
	}
	
	public int getID() {
		return id;
	}
	
	public void setName(String name) {
		this.name = name;
	}
	
	public String getName() {
		return name;
	}
	
	public void toggleSelection() {
		this.isSelected = !this.isSelected;
	}
	
	public void isSelected(boolean val) {
		this.isSelected = val;
	}
	
	public boolean isSelected() {
		return this.isSelected;
	}
	
	
	public void toggleHighlighting() {
		this.isHighlighted = !this.isHighlighted;
	}
	
	
	public void isHighlighted(boolean val) {
		this.isHighlighted = val;
	}
	
	public boolean isHighlighted() {
		return this.isHighlighted;
	}
	

	
}

File: cpp/peachtree/src/aln/Taxon.cpp
/*
 * Taxon.cpp
 *
 *  Created on: 30/06/2021
 *      Author: jdou557
 */

#include "Taxon.h"

Taxon::Taxon(int id, string name) {
	this->id = id;
	this->name = name;
	this->isHighlighted = false;
	this->isSelected = false;
	this->numInfections = -1;
}



int Taxon::getID() {
	return id;
}

void Taxon::setName(string name) {
	this->name = name;
}

string Taxon::getName() {
	return name;
}


string Taxon::getValue(string var){
	if (this->annotations.find(var) == this->annotations.end()) return "";
	return this->annotations[var];
}

void Taxon::setValue(string var, string val){
	this->annotations[var] = val;
}

void Taxon::toggleSelection() {
	this->isSelected = !this->isSelected;
}

void Taxon::setIsSelected(bool val) {
	this->isSelected = val;
}

bool Taxon::getIsSelected() {
	return this->isSelected;
}


void Taxon::toggleHighlighting() {
	this->isHighlighted = !this->isHighlighted;
}


void Taxon::setIsHighlighted(bool val) {
	this->isHighlighted = val;
}

bool Taxon::getIsHighlighted() {
	return this->isHighlighted;
}

void Taxon::setInfectionCount(double numInfections){
	this->numInfections = numInfections;
}

double Taxon::getInfectionCount(){
	return this->numInfections;
}


